MystenLabs/sui
# indexer: check-in various backfills

## Description 

The main motivation for this PR was to re-implement the `tx_affected_objects` using the ingestion framework, but it also includes a couple of chore (separated into their own commits, to be landed separately):

- Removing redundant curly braces on struct without fields.
- Reducing the visibility of crates (`pub(crate)` instead of `pub`).
- Fixing the default DB in checked in SQL backfill scripts (they were missing the default DB password).
- Checking in the `tx_affected_addresses` backfill that has already run.

## Test plan 

Testing the new `tx_affected_objects` backfill:

- Run the indexer locally, against a remote store.
- Create a copy of its `tx_affected_objects` table and create a fresh table to backfil into.
- Find the backfill high watermark:

```sql
SELECT MAX(sequence_number) FROM checkpoints;
```

- Run the backfill against the same remote store, between `0` and the high watermark.
- Confirm that the table you created with backfill has the same contents as the backup:

```sql
-- All of these should be the same count
SELECT COUNT(1) FROM tx_affected_objects;

SELECT COUNT(1) FROM tx_affected_objects_backup;

SELECT
    COUNT(1)
FROM
    tx_affected_objects
INNER JOIN
    tx_affected_objects_backup
USING (
    affected,
    tx_sequence_number,
    sender
);
```

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates.
